### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

File: doc/implementation_details.rst [hunk 1]
```diff
@@ -8,18 +8,117 @@ State machines
...
+* connect
+* release
+* DControl ("Parameter end" is sent to IO-Device)
+* CControl ("Application ready" is sent to IO-Controller)
+* parameter read (Uses CMRDR)
+* parameter write
 
+Incoming UDP packets are parsed by ``pf_cmrpc_dce_packet()``, which also
+prepares the return UDP packet. This is done by putting together incoming
+fragments and then call ``pf_cmrpc_rpc_request()``.
```
Comment (hefloryd) [inline doc/implementation_details.rst]: calling

File: doc/implementation_details.rst [hunk 2]
```diff
@@ -8,18 +8,117 @@ State machines
...
+ * ``pnet_exp_submodule_ind()``
+ * ``pnet_connect_ind()``
+ * ``pnet_state_ind()`` with PNET_EVENT_STARTUP
 
-* req
-* ind
-* rsp
-* cnf
\ No newline at end of file
+For incoming (parameter) write request messages is ``pf_cmrpc_rm_write_ind()`` called,
```
Comment (hefloryd) [inline doc/implementation_details.rst]: pf_cmrpc_rm_write_ind() is called for incoming ....

File: doc/creating_gsdml_files.rst [hunk 3]
```diff
@@ -57,12 +57,27 @@ The current GSD version (as of writing) is 2.35.
...
 The file name is specified in section 5.1 of the GSD specification. For example
 ``GSDML-V2.35-Vendor-Device-20171231.xml``. You can give it an optional timestamp
-in the filename: "-20171231-235900.xml”.
+in the filename: ``-20171231-235900.xml``.
 
 Note that version for example 2.3 is newer than version 2.25. From the GSDML
 specification: "V2.2 < V2.25 < V2.3 < V2.31"
 
 
+In-file comments
```
Comment (hefloryd) [inline doc/creating_gsdml_files.rst]: Do we need to explain basic XML concepts in this file?

File: include/pnet_api.h [hunk 4]
```diff
@@ -589,7 +589,8 @@ typedef int (*pnet_state_ind)(
  * value of \a idx (0x0000 - 0x7fff). All other values of \a idx are handled internally
  * by the Profinet stack.
  *
- * The application must verify the values of \a idx and \a write_length and provide a
+ * The application must verify the values of \a idx and that \a p_read_length is
```
Comment (hefloryd) [inline include/pnet_api.h]: must verify that

File: doc/implementation_details.rst (hunk 1 above)
Comment (pyhys) [reply doc/implementation_details.rst]: Done

File: doc/implementation_details.rst (hunk 2 above)
Comment (pyhys) [reply doc/implementation_details.rst]: Done

File: include/pnet_api.h (hunk 4 above)
Comment (pyhys) [reply include/pnet_api.h]: Changed to "The application must verify the value of \a idx, and that \a p_read_length is large enough."

File: doc/creating_gsdml_files.rst (hunk 3 above)
Comment (pyhys) [reply doc/creating_gsdml_files.rst]: Perhaps not, but many readers are more used to .EDS files where comments are different.

**Pull request merged**